dask/array/core.py

```diff
@@ -818,7 +818,7 @@ def map_blocks(
 
         dtype = apply_infer_dtype(func, args, original_kwargs, "map_blocks")
 
-    if drop_axis:
+    if drop_axis is not None:
```

Comment:
```suggestion
    if drop_axis:
```

Sorry for missing this earlier. I don't think `drop_axis` can ever be `None`, because the first thing we do in `map_blocks` is set `drop_axis` to `[]` if it is `None`.

Reply:
When I revert to
```
if drop_axis:
```
The test https://github.com/dask/dask/blob/8a6da1d9432a53359954386e4db915789f0163a0/dask/array/tests/test_routines.py#L2713
fails at https://github.com/dask/dask/blob/8a6da1d9432a53359954386e4db915789f0163a0/dask/array/tests/test_routines.py#L2731 when a 3-dimensional array is tested with the error
```pytr
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```
My previous fix was bad indeed: is the new commit acceptable?

Reply:
Ah, I see. A `len` check seems reasonable to me.